server/service/integration_core_test.go

```diff
@@ -799,6 +801,7 @@ func (s *integrationTestSuite) TestListHosts() {
 
 	user1 := test.NewUser(t, s.ds, "Alice", "alice@example.com", true)
 	q := test.NewQuery(t, s.ds, "query1", "select 1", 0, true)
+	defer cleanupQuery(s, q.ID)
```

Comment:
Should we use `t.Cleanup` instead?

Reply:
It's equivalent, I find `t.Cleanup` helpful when you want something done at the end of the test, but you're currently in a helper function and cannot defer it (so you don't have to return a cleanup function to defer to the caller), but when you're cleaning up something inside the test itself, I tend to use the standard language features.